### PR TITLE
[PKG-3361] more-itertools v10.1.0

### PR DESCRIPTION
**Pull request by lorepirri**
[PKG-3361] more-itertools 10.1.0

- upstream: https://github.com/more-itertools/more-itertools/tree/v10.1.0
- dependency files:
    - setup.cfg:  https://github.com/more-itertools/more-itertools/blob/v10.1.0/setup.cfg
    - setup.py:  https://github.com/more-itertools/more-itertools/blob/v10.1.0/setup.py
    -  pyproject.toml:  https://github.com/more-itertools/more-itertools/blob/v10.1.0/pyproject.toml
- license: https://github.com/more-itertools/more-itertools/blob/v10.1.0/LICENSE

- conda-forge: https://github.com/conda-forge/more-itertools-feedstock/blob/main/recipe/meta.yaml
- pypi: https://pypi.org/project/more-itertools/
    
**Destination channel:** `defaults`

**Changes**
- update to v10.1.0
- fix recipe

**Dependency for**
- [kedro v0.18.14 ❄️](https://anaconda.atlassian.net/browse/PKG-3345)

[PKG-3361]: https://anaconda.atlassian.net/browse/PKG-3361?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -31,16 +29,20 @@ test:
     - pip check
   requires:
     - pip
-    - python <3.10
+    - python
```
Comment (danpetry) [inline recipe/meta.yaml:32]: don't think python is required for pip check

**Review by danpetry — APPROVED** [review]
approved based on comment being addressed

**Comment by skupr-anaconda** [comment]
> * https://github.com/more-itertools/more-itertools/blob/v10.1.0/pyproject.toml

@lorepirri No need to provide setup.cfg or setup.py if they are don't have dependencies or other requirements

File: recipe/meta.yaml (hunk 1 above)
Comment (skupr-anaconda) [reply recipe/meta.yaml:32]: Correct. `python <3.10` was an old trick for python 3.10 build-out and must be removed
```suggestion
```

**Review by skupr-anaconda — APPROVED** [review]
LGTM but remove python from test/requires

**Pull request merged**